src/classlibnative/bcltype/system.cpp

```diff
@@ -288,11 +289,13 @@ FCIMPL0(StringObject*, SystemNative::_GetModuleFileName)
 {
     FCALL_CONTRACT;
 
-    WCHAR wszFile[MAX_LONGPATH];
+    PathString wszFilePathString;
+    
     STRINGREF   refRetVal   = NULL;
```

Comment:
There are very subtle rules that need to be following during implementation of FCall. The ones violated here are:
- stack allocated (ie `PathString` objects with constructors has to be inside `HELPER_METHOD_FRAME` scope
- exception has to be thrown inside `HELPER_METHOD_FRAME` scope (`GetUnicode()` can throw)